pkg/build/api/v1/types.go

```diff
@@ -371,6 +375,13 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// RunBash holds a bash script to be executed in a container running the
+	// build output image.
+	RunBash string `json:"runBash,omitempty" description:"bash script to be executed in a container running the build output image"`
```

Comment:
> This seems really free-form. Why not force some structure into this by allowing references to scripts in the repo itself?

It has been previously discussed with @smarterclayton and @bparees, and we explicitly did not want that approach.

The objective here is to allow a developer to run her project's unit tests as part of the build process with minimum amount of fuss. In particular, there should be no changes required to the source repository.

The experience we're looking for is to allow folks to say "to run my tests, run `rake test`" and OpenShift will run that as part of the build.

If the size of this string is a concern, we might limit it with validation?